be1-go/channel/chirp/mod.go

```diff
@@ -25,11 +26,10 @@ const failedToDecodeData = "failed to decode message data: %v"
 
 // NewChannel returns a new initialized individual chirping channel
 func NewChannel(channelPath string, ownerKey string, hub channel.HubFunctionalities,
-	generalChannel channel.Broadcastable, log zerolog.Logger) Channel {
+	generalChannel channel.Broadcastable, log zerolog.Logger) channel.Channel {
```

Comment:
I'm not convinced this is a good idea since we need to use this function and have the concrete type in the tests. Same for the GeneralChirp channel.